src/Types/IndexTypes.cs

```diff
@@ -48,17 +94,130 @@ public enum IndexState
     InitializationFailed = 7
 }
 
+/// <summary>
+/// Index specification.
+/// </summary>
+public record IndexSpec
+{
+    /// <summary>
+    /// Serverless index specification. <see langword="null" /> if the index is pod-based.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public ServerlessSpec? Serverless { get; init; }
+
+    /// <summary>
+    /// Pod-based index specification. <see langword="null" /> if the index is serverless.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public PodSpec? Pod { get; init; }
+}
+
+/// <summary>
+/// Serverless index specification.
+/// </summary>
+public record ServerlessSpec
+{
+    /// <summary>
+    /// The public cloud where the index is hosted.
+    /// </summary>
+    public required string Cloud { get; init; }
+
+    /// <summary>
+    /// The region where the index has been created.
+    /// </summary>
+    public required string Region { get; init; }
+}
+
+/// <summary>
+/// Pod-based index specification.
+/// </summary>
+public record PodSpec
+{
+    /// <summary>
+    /// The environment where the index is hosted.
+    /// </summary>
+    [JsonPropertyName("environment")]
+    public required string Environment { get; init; }
+
+    /// <summary>
+    /// The pod type.
+    /// </summary>
+    [JsonPropertyName("pod_type")]
+    public required string PodType { get; init; }
+
+    /// <summary>
+    /// The number of pods used.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public long? Pods { get; init; }
```

Comment:
`uint?` rather than 'long?' for pods, replicas and shards?

Reply:
Makes sense, it's unlikely anyone would have more than 4B pods, unless we're in a *really* fast AI takeoff scenario.